frontend/src/metabase/query_builder/components/filters/FilterPopover.jsx

```diff
@@ -147,6 +155,7 @@ export default class ViewFilterPopover extends Component {
           startRule="boolean"
           isValid={filter && filter.isValid()}
           onChange={this.handleFilterChange}
+          onUpdateAndCommit={this.handleUpdateAndCommit}
           onBack={() => this.setState({ editingFilter: false })}
           onDone={this.handleCommit}
```

Comment:
Concerning the distinction between "done" and "commit" it looks like the hierarchy of components goes `FilterPopover (handleCommit)` to `ExpressionPopover (onDone)` to `ExpressionEditorTextfield (onCommit)`. "done" here feels a bit redundant so how do you feel about merging `onDone` and `onUpdateAndCommit` into a single prop so that we can avoid passing down a fourth callback?

Reply:
Done! Please take a look again.